DOC: Improve ``np.mean`` documentation of the out argument (#25431)
numpy/numpy: doc/source/user/basics.ufuncs.rst

Finally, the *out* keyword allows you to
provide an output array (for single-output ufuncs, which are currently the only
ones supported; for future extension, however, a tuple with a single argument
can be passed in). If *out* is given, the *dtype* argument is only used for the

> internal computations.

Reviewer: How did you determine this?

> can be passed in). If *out* is given, the *dtype* argument is only used for the
> internal computations.

Reviewer: ```suggestion
can be passed in). If *out* is given, the *dtype* argument is ignored.
```
Reverting this change since https://github.com/numpy/numpy/pull/25431/files#r1451835427 is unresolved. Please open another PR if you wish to add details here, and ask for someone familiar with ufunc implementation to review.
